extensions/panache/panache-mock/src/main/java/io/quarkus/panache/mock/MockPanacheEntities.java

```diff
@@ -0,0 +1,24 @@
+package io.quarkus.panache.mock;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Install mocks for the given entity classes.
+ * <p>
+ * This annotation only works together with {@code QuarkusComponentTest}.
+ *
+ * @see PanacheMock#mock(Class...)
+ */
+@Retention(RUNTIME)
+@Target(TYPE)
+public @interface MockPanacheEntities {
```

Comment:
I suppose this is a useful alternative to `PanacheMock.mock(entity)` but then why not support it also in `@QuarkusTest`? Isn't this confusing?
Is it _required_ for `QuarkusComponentTest`?

Reply:
> I suppose this is a useful alternative to `PanacheMock.mock(entity)` but then why not support it also in `@QuarkusTest`? Isn't this confusing? 

Well, I didn't want to modify the way `PanacheMock` is used normally. `PanacheMock` also lacks some meaningful javadoc but that's out of the scope of this pull request. I guess that we could add a `QuarkusTestBeforeEachCallback` that would handle this annotation for `@QuarkusTest`. But I'd like to address this in a follow-up PR.

> Is it _required_ for `QuarkusComponentTest`?

Yes, it is. We need a declarative way to define entities that should be mocked.